trace/span.proto

```diff
@@ -0,0 +1,323 @@
+  // is the time when the server's application handler starts running.
+  google.protobuf.Timestamp start_time = 5;
+
+  // The end time of the span. On the client side, this is the time kept by
+  // the local machine where the span execution ends. On the server side, this
+  // is the time when the server application handler stops running.
+  google.protobuf.Timestamp end_time = 6;
+
+  // A set of attributes, each in the format `[KEY]:[VALUE]`.
+  message Attributes {
+    // The set of attributes. The value can be a string, an integer, or the
+    // Boolean values `true` and `false`. For example:
+    //
+    //     "/instance_id": "my-instance"
+    //     "/http/user_agent": ""
+    //     "/http/server_latency": 300
+    //     "abc.com/myattribute": true
+    map<string, AttributeValue> attribute_map = 1;
+
+    // The number of attributes that were discarded. Attributes can be discarded
+    // because their keys are too long or because there are too many attributes.
+    // If this value is 0 then all attributes are valid.
+    int32 dropped_attributes_count = 2;
+  }
+
+  // A set of attributes on the span.
+  Attributes attributes = 7;
+
+  // Stack trace captured at the start of the span.
+  StackTrace stack_trace = 8;
+
+  // A time-stamped annotation or network event in the Span.
+  message TimeEvent {
+    // The timestamp indicating the time the event occurred.
+    google.protobuf.Timestamp time = 1;
+
+    // Text annotation with a set of attributes.
+    message Annotation {
+      // A user-supplied message describing the event.
+      TruncatableString description = 1;
+
+      // A set of attributes on the annotation.
+      Attributes attributes = 2;
+    }
+
+    // An event describing an RPC message sent or received on the network.
+    message NetworkEvent {
+      // For sent messages, this is the time at which the first bit was sent.
+      // For received messages, this is the time at which the last bit was
+      // received.
+      google.protobuf.Timestamp time = 1;
+
+      // Indicates whether the network message was sent or received.
+      enum Type {
+        // Unknown event type.
+        TYPE_UNSPECIFIED = 0;
+        // Indicates a sent RPC message.
+        SENT = 1;
+        // Indicates a received RPC message.
+        RECV = 2;
+      }
+
+      // Type of NetworkEvent. Indicates whether the RPC message was sent or
+      // received.
+      Type type = 2;
+
+      // An identifier for the message, which must be unique in this span.
+      uint64 message_id = 3;
+
+      // The number of uncompressed bytes sent or received.
+      uint64 uncompressed_message_size = 4;
+
+      // The number of compressed bytes sent or received.
+      uint64 compressed_message_size = 5;
+    }
+
+    // A `TimeEvent` can contain either an `Annotation` object or a
+    // `NetworkEvent` object, but not both.
+    oneof value {
+      // One or more key:value pairs.
+      Annotation annotation = 2;
```

Comment:
"Text annotation with a set of attributes." as opposed to key/value pairs

Reply:
Good catch.